### ISSUE TITLE
Don't hardcode compiler

### ISSUE THREAD

**Issue by r-barnes** [changliao1025/hexwatershed_tutorial#3]
Your Mac setup instructions include
```
cmake CMakeLists.txt -DCMAKE_CXX_COMPILER=g++-11
```
Macs often, but not always, come installed with the clang compiler. Using the above directive causes cmake to ignore this perfectly functional compiler in favour of forcing the user to figure out how to install GCC. Dropping `-DCMAKE_CXX_COMPILER=g++-11` will cause cmake to automatically select an appropriate compiler if one is available.

**Comment by changliao1025** [comment]
Yes, we will test the clang compiler and relax this requirement.
